src/paging.rs

```diff
@@ -653,12 +661,8 @@ pub struct Descriptor(usize);
 impl Descriptor {
     const PHYSICAL_ADDRESS_BITMASK: usize = !(PAGE_SIZE - 1) & !(0xffff << 48);
 
-    fn output_address(self) -> Option<PhysicalAddress> {
-        if self.is_valid() {
-            Some(PhysicalAddress(self.0 & Self::PHYSICAL_ADDRESS_BITMASK))
-        } else {
-            None
-        }
+    fn output_address(self) -> PhysicalAddress {
```

Comment:
Should this be named `physical_address`, for clarity? Or even be `impl From<Descriptor> for PhysicalAddress` (not sure why it wasn't already `TryFrom`)?

Reply:
I think the current name is clearer; it's not just any physical address, it's specifically the output address of the page table mapping.